spec/ddtrace/transport/http_spec.rb

```diff
@@ -45,14 +49,21 @@
 
       default.apis.each_value do |api|
         expect(api).to be_a_kind_of(Datadog::Transport::HTTP::API::Instance)
-        expect(api.adapter).to be_a_kind_of(Datadog::Transport::HTTP::Adapters::Net)
-        expect(api.adapter.hostname).to \
-          eq(Datadog::Configuration::AgentSettingsResolver::ENVIRONMENT_AGENT_SETTINGS.hostname)
-        expect(api.adapter.port).to eq(Datadog::Configuration::AgentSettingsResolver::ENVIRONMENT_AGENT_SETTINGS.port)
-        expect(api.adapter.timeout).to \
-          eq(Datadog::Configuration::AgentSettingsResolver::ENVIRONMENT_AGENT_SETTINGS.timeout_seconds)
-        expect(api.adapter.ssl).to be(Datadog::Configuration::AgentSettingsResolver::ENVIRONMENT_AGENT_SETTINGS.ssl)
+        expect(api.adapter.timeout).to eq(env_agent_settings.timeout_seconds)
         expect(api.headers).to include(described_class.default_headers)
+
+        case env_agent_settings.adapter
+        when :net_http
+          expect(api.adapter).to be_a_kind_of(Datadog::Transport::HTTP::Adapters::Net)
+          expect(api.adapter.hostname).to eq(env_agent_settings.hostname)
+          expect(api.adapter.port).to eq(env_agent_settings.port)
+          expect(api.adapter.ssl).to be(env_agent_settings.ssl)
+        when :unix
+          expect(api.adapter).to be_a_kind_of(Datadog::Transport::HTTP::Adapters::UnixSocket)
+          expect(api.adapter.filepath).to eq(env_agent_settings.uds_path)
+        else
+          raise("Unknown default adapter: #{env_agent_settings.adapter}")
+        end
       end
```

Comment:
Minor: Hmm... This is getting kinda hairy. I wonder if we should just remove the `ENVIRONMENT_AGENT_SETTINGS` default; is there still any production code path when it would be used, or is it just a tests thing?

Reply:
`ENVIRONMENT_AGENT_SETTINGS` can be removed, bit at quick glance it deserves its own PR.

Reply:
Ack! Do you mind leaving a comment marking it as deprecated so we don't forget? :)

Reply:
Sure, added a comment there.